cmd/pebble/cmd_pull.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/canonical/pebble/client"
+	"github.com/canonical/pebble/internal/progress"
+	"github.com/canonical/pebble/internal/strutil/quantity"
+)
+
+type cmdPull struct {
+	clientMixin
+
+	Positional struct {
+		RemotePath string `positional-arg-name:"<remote-path>" required:"1"`
+		LocalPath  string `positional-arg-name:"<local-path>" required:"1"`
+	} `positional-args:"yes"`
+}
+
+var shortPullHelp = "Retrieve a file from the remote system"
+var longPullHelp = `
+The pull command retrieves a file from the remote system.
+`
+
+type pullProgress struct {
+	file    *os.File
+	size    int64
+	current int64
+	pb      progress.Meter
+	started time.Time
+	msg     string
+}
+
+func (p *pullProgress) Write(b []byte) (n int, err error) {
+	n, err = p.file.Write(b)
+	if err != nil {
+		return
+	}
+
+	p.current += int64(n)
+	p.pb.Set(float64(p.current))
+
+	if p.current == p.size {
+		p.pb.Finished()
+
+		size := quantity.FormatAmount(uint64(p.size), 0)
+		duration := quantity.FormatDuration(time.Since(p.started).Seconds())
+		p.pb.Notify(fmt.Sprintf("Transferred %sB in %s", size, duration))
+	}
+
+	return
+}
+
+func newPullProgress(f *os.File, remotePath string) (*pullProgress, error) {
```

Comment:
Similar comment here as below -- maybe we leave the progress bar to a separate PR to keep this one smaller (and maybe not do it at all). It also requires the new `Content-Length` header you've added, which won't be available on older servers (see related comments).